### ISSUE TITLE
Adding code example about Physics-Informed Neural Networks for solving Eikonal equation

### ISSUE THREAD

**Issue by sgrubas** [keras-team/keras-io#1287]
Hello,

I would like to contribute to the [Keras examples](https://keras.io/examples/) by adding a code example implementing Physics-Informed Neural Networks (PINNs).

This example would be focused on solving the Eikonal equation via implementation of the [Neural Eikonal Solver](https://github.com/sgrubas/NES) which is based on [recent research](https://doi.org/10.1016/j.jcp.2022.111789) proposing fast and robust neural-network solution to the equation. The Eikonal equation is a nonlinear Partial Differential Equation (PDE) and it might be of interest to a wide audience because it is used in 3D rendering and computer vision via Signed Distance Functions, as well as commonly used in seismology. The proposed code example will also demonstrate the usage of AutoDiff operations (`tf.gradients`) inside the trainable models to solve the PDE. 

The example would contain the implementation of the [NES](https://github.com/sgrubas/NES) architecture and its training on one example with a complex model. The training of ~1-2 min will be enough to provide a very accurate solution.

To date, the [Keras](https://keras.io/examples/) repository does not have examples about Physics-Informed Neural Networks. 

Please, let me know if this example could be a valuable contribution to the Keras repository.

Thank you for your time.

**Comment by arre-ankit** [comment]
Yeah, I would like to contribute in Physics Informed Neural Network and I am also currently working on project on PIINs in my college i would like to know the possibilities of PIINs in different domains

**Comment by fchollet** [comment]
That sounds great! Please feel free to open a PR. I think it would make a valuable addition.
